### PR TITLE
Add clippy lints to the project, address some dings 

### PR DESCRIPTION
**Pull request by blt**
This commit is a follow-on from a suggestion I had early on to make aggressive use of clippy. The changes here are mostly mechanical however some needless clones, branches have been caught and fixed. I have left some of the precision loss dings in place, note that lib.rs has a section of lints explicitly allowed. These should be an hour or two to grind out. Some of these lints are also a matter of project taste. 

Assumes that https://github.com/DataDog/datadog-lambda-extension/pull/227 will merge. Should be rebased once that PR is in `main`.

### CODE REVIEW

File: bottlecap/src/telemetry/listener.rs [hunk 1]
```diff
@@ -116,13 +122,18 @@ pub struct TelemetryListenerConfig {
 }
 
 impl TelemetryListener {
+    /// Run the `TelemetryListener`
+    ///
+    /// # Errors
+    ///
+    /// Fucntion will error if the passed address cannot be bound.
```
Comment (duncanista) [inline bottlecap/src/telemetry/listener.rs:129]:
```suggestion
    /// Function will error if the passed address cannot be bound.
```

File: bottlecap/src/config.rs [hunk 2]
```diff
@@ -68,36 +69,32 @@ impl<'de> Deserialize<'de> for FlushStrategy {
...
+                        Ok(FlushStrategy::Default)
+                    } else {
                         debug!("invalid flush strategy: {}, using default", value);
                         Ok(FlushStrategy::Default)
                     }
                 }
+                _ => {
+                    debug!("invalid flush strategy: {}, using default", value);
+                    Ok(FlushStrategy::Default)
+                }
```
Comment (duncanista) [inline bottlecap/src/config.rs:97]: Thanks for improving this

**Review by duncanista — APPROVED** [review]
LGTM – left a nit.

File: bottlecap/src/config.rs (hunk 2 above)
Comment (blt) [reply bottlecap/src/config.rs:97]: 100% a clippy lint fix.

**Pull request merged**